example_allocators/bump_allocator/bump_alloc.c

```diff
@@ -0,0 +1,80 @@
+/**********************************
+ * bump_alloc.c
+ * Jeremy.Singer@glasgow.ac.uk
+ *
+ * This is a simple bump-pointer allocator.
+ * It mmaps a large buffer of SIZE bytes,
+ * then allocates this space in word-sized
+ * chunks to client code (in main fn).
+ *
+ * Initial simple memory allocator test.
+ * Explore capability narrowing operations
+ * and intrinsics for bound reporting.
+ */
+
+#include <cheriintrin.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+
+#include "bump_alloc.h"
+
+int count = 0;		 /* number of bytes allocated so far*/
+int max = 0;		 /* upper limit for count */
+char *buffer = NULL; /* the allocation buffer */
+
+void init_alloc(int size_in_bytes)
+{
+	/* request memory for our allocation buffer */
+	char *res = mmap(NULL, size_in_bytes, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1, 0);
+
+	if (res == MAP_FAILED)
+	{
+		perror("error in initial mem allocation");
+		exit(-1);
+	}
+
+	/* NB mmap min bounds for capability is 1 page (4K) */
+	// if (DEBUG_PRINTF)
```

Comment:
We should probably remove commented code?

Reply:
fixed in 6b7294a